docs/src/passing-in-data.rst

```diff
@@ -0,0 +1,190 @@
+   limitations under the License.
+
+.. _passing_in_data:
+
+##########################################
+Passing in External Data to Your |cdk| App
+##########################################
+
+.. See https://github.com/awslabs/aws-cdk/issues/603 (includes work from the following PR)
+       https://github.com/awslabs/aws-cdk/pull/183
+
+There are a number of ways you can get data into your app from outside your app.
+
+- Using a command-line context variable
+- Using an environment variable
+- Using an SSM Parameter Store variable
+- Using a value from another stack
+- Using a |CFN| template value
+
+Each of these techniques is described in the following sections.
+
+.. _passing_in_data_from_command_line:
+
+Getting a Value from a Command-Line Context Variable
+====================================================
+
+To pass the value of a command-line context value to your app,
+run a |toolkit| command using the **--context** (**-c**) option,
+as shown in the following example.
+
+.. code-block:: sh
+
+    cdk synth -c bucket_name=mygroovybucket
+
+To get the value of a command-line context value in your app,
+use code like the following,
+which gets the value of the context variable **bucket_name**.
+
+.. code-block:: ts
+
+    const bucket_name string = this.getContext("bucket_name");
+
+.. _passing_in_data_from_env_vars:
+
+Getting a Value from an Environment Variable
+============================================
+
+To get the value of an environment variable,
+use code like the following,
+which gets the value of the environment variable **MYBUCKET**.
+
+.. code-block:: ts
+
+    const bucket_name = process.env.MYBUCKET;
+
+.. _passing_in_data_from_ssm:
+
+Getting a Value from an SSM Store Variable
+==========================================
+
+To get the value of an SSM parameter store variable,
+use code like the following,
+which uses the value of the SSM variable **my-awesome-value**.
+
+.. code-block:: ts
+		
+    const myvalue = new SSMParameterProvider(this).getString("my-awesome-value");
+
+See the :doc:`context` topic for information about the :py:class:`SSMParameterProvider <@aws-cdk/cdk.SSMParameterProvider>`.
+
+.. _passing_in_data_between_stacks:
+
+Passing in Data From Another Stack
+==================================
+
+You can pass data from one stack to another stack in the same app
+by using the **export** method in one stack and the **import** method in the other stack.
+
+The following example creates a bucket on one stack
+and passes a reference to the other stack through an interface.
+
+First create a stack with a bucket.
+The stack includes a a property we use to pass the bucket's properties to the other stack.
+Note how we use the **export** method on the bucket to get it's properties and save them
+in the stack property.
+
+.. code-block:: ts
+
+    class HelloCdkStack extends cdk.Stack {
+        // Property that defines the stack you are exporting from
```

Comment:
Not sure about this comment. What does it mean?

Reply:
I'll revisit.